modules/core/src/layers/xviz-layer.js

```diff
@@ -107,6 +107,129 @@ const STYLE_TO_LAYER_PROP = {
     text_baseline: 'getAlignmentBaseline'
   }
 };
+/* 
+ ####### EXTENDING XVIZ LAYER FOR PRIMITIVES #########
+*/
+// not sure if necessary, just to have enums for existing primitives
+const LAYER_TYPES = {
```

Comment:
I'd really like to find a way to keep this at the XVIZLayer type, not the deck.gl type, but i'm not sure that makes sense considering we are "extending".  You have to know the underlying type, but I feel that limits streetscape being able to change that at any time?

Given that, maybe custom layers like this could be excluded from the semver policy (not sure where to put this), but we should probably ensure we have some way to signal that?